sale_substate/views/sale_views.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 Akretion
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo>
+    <record id="sale_order_view_form" model="ir.ui.view">
+      <field name="model">sale.order</field>
+      <field name="inherit_id" ref="sale.view_order_form" />
+      <field name="arch" type="xml">
+          <xpath expr="//header/field[@name='state']" position="before">
```

Comment:
```
<xpath expr="//header" position="after">
    <header name="substate">
        <field name="substate_id" widget="statusbar" options="{'clickable': '1',}"/>
    </header>
</xpath>
```

Is it posible to create new header for substate?

Reply:
I think it is the matter of the implementation, you can decide yourself.
This module is just the example, right? @mourad-ehm

Reply:
Yes, because UI is also depending on number of substates. In most cases we don't need all this substates. For ex Negociation steps can be managed in crm. Delivered sub-state in most case is not necessary.

Reply:
@Saran440,  I tested your proposition it's ok form me. Could you make a commit suggest ?

Reply:
@mourad-ehm I concern that there are 2 header in form will bug later if other module inherit header. or what do you think?

Reply:
I think that there is no problem. The inheritance get the original view unless we change priority (which can be done by user for a specific use). Moreover, the header has a name.

Reply:
okay, I commit pr to https://github.com/akretion/sale-workflow/pull/19